pkg/review/commit_review_status.go

```diff
@@ -91,6 +106,71 @@ type PullRequest struct {
 	ReviewDecision githubv4.String
 }
 
+// CommitApprovalPipelineConfig holds the configuration data for the
+// commit approval beam pipeline.
+type CommitApprovalPipelineConfig struct {
+	// The token to use for authenticating with GitHub.
+	GitHubAccessToken string
+	// The GCP project that holds the BigQuery tables
+	BigQueryProject string
+	// The name of the BigQuery dataset that contains the tables
+	BigQueryDataset string
+	// The name of the BigQuery table that holds push event data. This is the
+	// table that is used to source the commits that need to be processed.
+	BigQueryPushEventsTable string
+	// The name of the BigQuery table that holds the commit review/approval status.
+	// This is the table that stores the final output of pipeline.
+	BigQueryCommitReviewStatusTable string
+	// The name of the BigQuery table that holds GitHub issue data. This table
+	// is used to determine if a commit was pushed using 'break glass' permissions.
+	BigQueryIssuesTable string
+}
+
+// CommitApprovalDoFn is an object that implements beams "DoFn" interface to
+// provide the processing logic for converting a Commit to CommitReviewStatus.
+type CommitApprovalDoFn struct {
+	Config       CommitApprovalPipelineConfig
+	GithubClient *githubv4.Client
+}
+
+// ProcessElement is a DoFn implementation that take a Commit, determines
+// if the commit was properly approved, and outputs the resulting
+// CommitReviewStatus using the provided emit function.
+// A commit is considered properly reviewed as long as there is an associated
+// PR for the commit targeting the repository's main branch with reviewDecision
+// of 'APPROVED'.
+func (fn *CommitApprovalDoFn) ProcessElement(ctx context.Context, commit Commit, emit func(CommitReviewStatus)) {
+	logger := logging.FromContext(ctx)
+	logger.Infow("processing commit", "commit", commit)
```

Comment:
I think we prefer InfoContext over Info. https://github.com/abcxyz/readability/wiki/Go-Best-Practices#logging

Same throughout this file; FooContext() vs Foo()

Reply:
the version of abcxyz/pkg that was being used was outdated. I've opened a PR to update it: #148. Once that's merged I will rebase this branch and updatre these logging statements.

Reply:
This has been fixed.